src/cli/cli_dataset.cpp

```diff
@@ -355,5 +355,208 @@ ThreadError Dataset::ProcessPendingTimestamp(int argc, char *argv[])
     return error;
 }
 
+ThreadError Dataset::ProcessMgmtCommand(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+    otOperationalDataset dataset;
+    uint8_t tlvs[32];
+    uint8_t size = 0;
+    uint8_t index;
+    long value;
+    int length;
+    otIp6Address prefix;
+
+    VerifyOrExit(argc > 0, error = kThreadError_Parse);
+
+    index = 2;
+
+    if (strcmp(argv[0], "get") == 0)
+    {
+        while (index < argc)
+        {
+            if (strcmp(argv[index], "activetimestamp") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_ACTIVETIMESTAMP;
+            }
+
+            if (strcmp(argv[index], "pendingtimestamp") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_PENDINGTIMESTAMP;
+            }
+
+            if (strcmp(argv[index], "masterkey") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_MASTERKEY;
+            }
+
+            if (strcmp(argv[index], "networkname") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_NETWORKNAME;
+            }
+
+            if (strcmp(argv[index], "extpanid") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_EXTPANID;
+            }
+
+            if (strcmp(argv[index], "localprefix") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_LOCALPREFIX;
+            }
+
+            if (strcmp(argv[index], "delaytimer") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_DELAYTIMER;
+            }
+
+            if (strcmp(argv[index], "panid") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_PANID;
+            }
+
+            if (strcmp(argv[index], "channel") == 0)
+            {
+                tlvs[size++] = OT_MESHCOP_TLV_CHANNEL;
+            }
+
+            if (strcmp(argv[index], "binary") == 0)
+            {
+                VerifyOrExit((index + 1) < argc, error = kThreadError_Parse);
+                SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
+                VerifyOrExit(Interpreter::Hex2Bin(argv[index + 1], tlvs + size, static_cast<uint16_t>(value)) >= 0,
+                             error = kThreadError_Parse);
+                size += value;
+            }
```

Comment:
Check for parsing errors?  For example, an arg that does not match any of the cases above?